### DISCUSSION TITLE
why City drive incompatible?

### DISCUSSION THREAD

**Discussion by lokpro** [CitiesSkylinesMods/TMPE#1438] [General]
City drive is on TMPE's incompatible-list.

City drive also says incompatible with TMPE on their page:
Steam Workshop::City Drive
https://steamcommunity.com/sharedfiles/filedetails/?id=844180955

I have been playing with both mods, and yet I did not notice anything wrong.

**Comment by krzychu124** [comment]
> I have been playing with both mods, and yet I did not notice anything wrong.

Have you played with TM:PE without CityDrive?
That mod is super old and its patches are completely replacing game logic, TM:PE will patch game functions without errors but those functions will never be invoked because of mentioned full replacement. These days we use Harmony to patch the same game function but without replacement so all mods patching it can work together (most of the times).

In short:
- `ParkingAI` won't work and may cause a lot of issues,
- Priority signs won't work correctly (TM:PE won't be able to track vehicles in some situations, so priority check result might be completely wrong e.g.: vehicle should yield but will not do that)
- disable vehicle despawn feature won't work for passenger vehicles, busses and cargo trucks (including post van and other service vehicles),
- Traffic flow info dependent features like _Dynamic Lane Selection_ won't work correctly, pathfinding may also find non-optimal paths (you may notice increased number of situations where on highway a lot of vehicles will switch lane to the same one)

**Comment by lokpro** [comment]
Hi, thanks to your insight reply, I realised that I have been using a defective TM:PE due to me putting it together with City Drive.

I installed City Drive first, TM:PE later. So, I always used TM:PE with City Drive.
I have noticed many weird routings (using small streets instead of main roads) all over my city, and this had lead me checking issues and codes for solutions...

Today, when I loaded my game without City Drive, I saw my traffic gradually changing from their old bad routes to normal, rational routes. Many stuck roads are now better. Thank you.
